lib/src/r_tree/node.dart

```diff
@@ -94,23 +87,21 @@ abstract class Node<E> implements RTreeContributor {
 
   /// Adds the rectangle containing [item] to this node's covered rectangle
   include(RTreeContributor item) {
-    _minimumBoundingRect = _minimumBoundingRect == null ? item.rect : rect.boundingBox(item.rect);
+    _minimumBoundingRect = _minimumBoundingRect == noMBR ? item.rect : rect.boundingBox(item.rect);
   }
 
   /// Recalculated the bounding rectangle of this node
   Rectangle updateBoundingRect() {
     if (children.isEmpty) {
-      _minimumBoundingRect = null;
-      return noMBR;
+      return _minimumBoundingRect = noMBR;
     }
 
-    var newBoundingRect = children.first.rect;
-    for (var child in children.skip(1)) {
-      newBoundingRect = newBoundingRect.boundingBox(child.rect);
+    var updatedBoundingRect = children[0].rect;
+    for (var i = 1; i < children.length; i++) {
+      updatedBoundingRect = updatedBoundingRect.boundingBox(children[i].rect);
     }
 
-    _minimumBoundingRect = newBoundingRect;
-    return _minimumBoundingRect!;
+    return _minimumBoundingRect = updatedBoundingRect;
```

Comment:
Same comment as above, let's split this logic into two separate lines.

Reply:
I think that's a good call, done